packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -522,65 +523,78 @@ IFluidDataStoreChannel, IFluidDataStoreRuntime, IFluidHandleContext {
-                        deferred.resolve(remoteChannelContext);
-                        this.contextsDeferred.set(id, deferred);
+                        assert(!this.contexts.has(id), `Unexpected attach channel OP,
+                            is in pendingAttach set: ${this.pendingAttach.has(id)},
+                            is local channel contexts: ${this.contexts.get(id) instanceof LocalChannelContext}`);
+
+                        const flatBlobs = new Map<string, string>();
+                        const snapshotTree = buildSnapshotTree(attachMessage.snapshot.entries, flatBlobs);
+
+                        const remoteChannelContext = new RemoteChannelContext(
+                            this,
+                            this.dataStoreContext,
+                            this.dataStoreContext.storage,
+                            (content, localContentMetadata) => this.submitChannelOp(id, content, localContentMetadata),
+                            (address: string) => this.setChannelDirty(address),
+                            id,
+                            snapshotTree,
+                            this.sharedObjectRegistry,
+                            flatBlobs,
+                            this.dataStoreContext.getCreateChildSummarizerNodeFn(
+                                id,
+                                {
+                                    type: CreateSummarizerNodeSource.FromAttach,
+                                    sequenceNumber: message.sequenceNumber,
+                                    snapshot: attachMessage.snapshot,
+                                },
+                            ),
+                            async () => this.getChannelInitialGCDetails(id),
+                            attachMessage.type);
+
+                        this.contexts.set(id, remoteChannelContext);
+                        if (this.contextsDeferred.has(id)) {
+                            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                            this.contextsDeferred.get(id)!.resolve(remoteChannelContext);
+                        } else {
+                            const deferred = new Deferred<IChannelContext>();
+                            deferred.resolve(remoteChannelContext);
+                            this.contextsDeferred.set(id, deferred);
+                        }
                     }
+                    break;
                 }
-                break;
+
+                case DataStoreMessageType.ChannelOp:
+                    this.processChannelOp(message, local, localOpMetadata);
+                    break;
+                default:
             }
 
-            case DataStoreMessageType.ChannelOp:
-                this.processChannelOp(message, local, localOpMetadata);
-                break;
-            default:
+            this.emit("op", message);
+        } catch (error) {
+            throw CreateCorruptionError(error, {
```

Comment:
Are we adding try/catches on every single layer? It looks like duplication of effort...